drivers/touchscreen/touchscreen.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2016 Linaro Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <touchscreen.h>
+
+/*
+ * For three touchscreen coordinates:
+ *
+ *   (T[0].x, T[0].y), (T[1].x, T[1].y) and (T[2].x, T[2].y)
+ *
+ * the corresponding display coordinates:
+ *
+ *   (D[0].x, D[0].y), (D[1].x, D[1].y) and (D[2].x, D[2].y)
+ *
+ * can be calculated by
+ *
+ * ( D[0].x )     ( T[0].x T[0].y 1 )     ( A )
+ * ( D[1].x )  =  ( T[1].x T[1].y 1 )  X  ( B )
+ * ( D[2].x )     ( T[2].x T[2].y 1 )     ( C )
+ *
+ * ( D[0].y )     ( T[0].x T[0].y 1 )     ( D )
+ * ( D[1].y )  =  ( T[1].x T[1].y 1 )  X  ( E )
+ * ( D[2].y )     ( T[2].x T[2].y 1 )     ( F )
+ *
+ * The unknowns A though F can be calculated by
+ *
+ * ( A )      ( D[0].x )
+ * ( B ) = M' ( D[1].x )
+ * ( C )      ( D[2].x )
+ *
+ * ( D )      ( D[0].y )
+ * ( E ) = M' ( D[1].y )
+ * ( F )      ( D[2].y )
+ *
+ * where M is the matrix
+ *
+ * ( T[0].x T[0].y 1 )
+ * ( T[1].x T[1].y 1 )
+ * ( T[2].x T[2].y 1 )
+ *
+ * and M' is the inverse of M, given by
+ *
+ * M' = 1/det(M) Adj(M)
+ *
+ */
+
+/*
+ * Shift value to scale calculations by to get greater precision but without
+ * risking overflow.
+ */
+static const int xlat_scale = 12;
+
+/*
+ * Initialise translation constants using 3 touchscreen coordinates which
+ * correspond to known display coordinates. These constants will be used by
+ * subsequent calls to touchscreen_translate to translate arbitrary points.
+ */
+void touchscreen_set_calibration(struct touchscreen_xlat *xlat,
+				 const struct touchscreen_point display[3],
+				 const struct touchscreen_point touchscreen[3])
+{
+	/* Use shorter aliases for coords */
+	const struct touchscreen_point *T = touchscreen;
```

Comment:
I'm not sure if maintainers would agree with usage of all capitals for something else but macros (variable name in this case).